intra_process_demo/src/cyclic_pipeline/cyclic_pipeline.cpp

```diff
@@ -29,7 +29,7 @@ struct IncrementerPipe : public rclcpp::Node
     // Create a subscription on the input topic.
     sub = this->create_subscription_with_unique_ptr_callback<std_msgs::msg::Int32>(
       in, rmw_qos_profile_default,
-      [this](std_msgs::msg::Int32::UniquePtr & msg) {
+      [this](std_msgs::msg::Int32::UniquePtr msg) {
```

Comment:
I've made the changes to not pass the `this` pointer in https://github.com/ros2/demos/tree/avoid-this-lambda you can merge it or if you'd rather not, I can create a separate PR for it.